src/filesystem/node/node-filesystem.ts

```diff
@@ -47,11 +47,15 @@ export class FileSystemNode implements FileSystem {
         if (!stat.isDirectory) {
             throw new Error(`File system root should point to a directory location. URI: ${this.rootUri}.`);
         }
+
+
         const rootPath = FileUri.fsPath(this.rootUri);
-        this.watcher = chokidar.watch(rootPath).on("all", (eventType: EventType, filename: string) => {
+        this.watcher = chokidar.watch(rootPath, {
+            awaitWriteFinish: true
+        }).on("all", (eventType: EventType, filename: string) => {
             if (this.client) {
                 const changeType = this.getFileChangeType(eventType);
-                if (changeType && typeof filename === "string") {
+                if (changeType !== undefined && typeof filename === "string") {
                     const segments = path.normalize(path.relative(rootPath, filename));
                     const changedUri = segments === "." ? rootUri : FileUri.create(path.resolve(rootPath, segments));
                     const change = new FileChange(changedUri.toString(), changeType);
```

Comment:
For my curiosity what happens if getFileChangeType throws ? it goes up to root level with the exception unhandled ?